src/tests/helpers.ts

```diff
@@ -67,7 +68,7 @@ export async function expectResolutionErrorCode(
   return callback.then(
     () => fail("Expected resolution error to be thrown but wasn't"),
     (error) => {
-      if (error instanceof ResolutionError && error.code === code) {
+      if (error instanceof ResolutionError || ConfigurationError && error.code === code) {
```

Comment:
It doesn't work like that. It should be `error instanceof ResolutionError || error instanceof ConfigurationError`.
But my question is: how did add it here without checking if it is working?

Reply:
I was surprised as well, but once I added that the tests to check for Configuration Error start to pass. I added 2-3 of them where I expect a Configuration error. So maybe typescript is smart enough to understand that?

Reply:
@JohnnyJumper TypeScript evaluates it the same way as JavaScript (it's easy to check in the compiled code).
The current version of that boolean expression is read as follows:
`error instanceof ResolutionError` OR (`ConfigurationError` (always evaluates to true) AND `error.code === code`)

which equals `error instanceof ResolutionError || error.code === code`

It would be practically the same logical mistake as you would have written `a === 1 || 2`

Reply:
Oh... Now I see it!  Thanks @mvlabat  !